src/control/server/engine/config_test.go

```diff
@@ -1162,6 +1162,54 @@ func TestConfig_UpdatePMDKEnvarsStackSizeDCPM(t *testing.T) {
 	}
 }
 
+func TestConfig_UpdateABTEnvarsUCX(t *testing.T) {
+	validConfig := func() *Config {
+		return MockConfig().
+			WithFabricProvider("ucx+ud_x")
+	}
+
+	for name, tc := range map[string]struct {
+		cfg                   *Config
+		expErr                error
+		expABTthreadStackSize int
+	}{
+		"empty config should not fail": {
+			cfg:                   MockConfig().WithFabricProvider("ucx+ud_x"),
+			expABTthreadStackSize: minABTThreadStackSizeUCX,
+		},
+		"valid config for UCX should not fail": {
+			cfg:                   validConfig().WithEnvVarAbtThreadStackSize(minABTThreadStackSizeUCX),
+			expABTthreadStackSize: minABTThreadStackSizeUCX,
+		},
+		"config for UCX without thread size should not fail": {
+			cfg:                   validConfig(),
+			expABTthreadStackSize: minABTThreadStackSizeUCX,
+		},
+		"config for UCX  with stack size big enough should not fail": {
+			cfg: validConfig().
+				WithEnvVarAbtThreadStackSize(minABTThreadStackSizeUCX + 1),
+			expABTthreadStackSize: minABTThreadStackSizeUCX + 1,
+		},
```

Comment:
Would be good to have a test case for the env var being set too small, since that takes a different code branch.

Reply:
It turns out it's not possible (or perhaps just not readily accessible) to correct a un-allowed value. Hence the config code now disallows this (as is done for the DCPM case) and a test now verifies this behavior.